swagger_parser/lib/src/config/yaml_config.dart

```diff
@@ -14,7 +15,24 @@ import 'config_exception.dart';
 /// attempts to get config from pubspec.yaml
 /// If that fails, throws an exception
 final class YamlConfig {
-  YamlConfig(List<String> arguments) {
+  /// Applies parameters directly from constructor
+  const YamlConfig({
+    required this.schemaFilePath,
```

Comment:
Please remove required for all fields in constructor

Reply:
required in the constructor makes it mandatory to pass all parameters to the factory constructor, which can prevent errors when adding new configuration parameters when fromYamlFile does not pass all parameters to YamlConfig. In general, this constructor can be made private, since it is only used for internal purposes.